kamon-akka-remote-2.4.x/src/main/scala/kamon/akka/instrumentation/RemotingInstrumentation.scala

```diff
@@ -46,7 +110,16 @@ class RemotingInstrumentation {
     ackOption foreach { ack ⇒ ackAndEnvelopeBuilder.setAck(ackBuilder(ack)) }
     envelopeBuilder.setMessage(serializedMessage)
 
-    envelopeBuilder.setTraceContext(remoteTraceContext)
+    if(Kamon.currentContext() != null) {
```

Comment:
is this nullcheck really necessary or leftover from when we didn't process system messages properly?